entrypoint.sh

```diff
@@ -26,18 +28,25 @@ if [ "$APPLICATION_TYPE" = "compliance-backend" ]; then
 
 	export RACK_ENV=${RACK_ENV:-"production"}
 
+    if isClowderEnabled; then
+        PORT=`ClowderConfigWebPort`
+    else
+        PORT="8080"
+    fi
+
 	if is_puma_installed; then
 		export_vars=$(cgroup-limits) ; export $export_vars
 
-		exec bundle exec "puma --config ../etc/puma.cfg"
+		exec bundle exec "puma --config ../etc/puma.cfg -b tcp://0.0.0.0:${PORT}"
```

Comment:
Observation: unquoted `PORT`. It is possible to drop the quotes within the exec and quote just the bind param.

Reply:
Shellcheck does not complain because it is already quoted. The reason for that rule, SC2046, is to prevent word splitting, which this type of quoting prevents.

Reply:
shellcheck likely does not complain because it is all quoted. I guess the `exec bundle exec` splits it itself, hence the problem stays.
```
$ myvar="echo test"
$ bundle exec "$myvar"
test
```

Reply:
Your example doesn't actually demonstrate the code in question:

```sh
$ MSG="foo bar baz"
$ bundle exec "echo $MSG"
foo bar baz
$ bundle exec "echo ${MSG}"
foo bar baz
```

I think we're good here. It's also worth noting that the port will never contain spaces anyway, so even if the words were split, there would be no issue.

Reply:
I guess all it needs another pair of quotes:
```
exec bundle exec "puma --config ../etc/puma.cfg -b tcp://0.0.0.0:'${PORT}'"
```
The double ones are for bundle exec, and the single ones for the subshell.

Here is a better example to get the idea of its power:
```
$ testvar="\$(touch /tmp/testfile)"
$ echo $testvar
$(touch /tmp/testfile)
$ be "$testvar"
$ ls /tmp/testfile 
/tmp/testfile
$ be "'$testvar'"
sh: line 1: $(touch /tmp/testfile): No such file or directory
```
I'm fine keeping as is, some hardening would not hurt :slightly_smiling_face:.